File: touch.js
const fs = require('fs');
const path = require('path');

const args = process.argv.slice(2);
let currentPath = process.cwd();

function createFile(filePath) {
    if (!fs.existsSync(filePath)) {
        fs.writeFileSync(filePath, '');
    }
}

function createDir(dirPath) {
    if (!fs.existsSync(dirPath)) {
        fs.mkdirSync(dirPath, { recursive: true });
    }
}

for (const name of args) {
    if (name === "..") {
        currentPath = path.dirname(currentPath);
    } else if (name.includes(".")) {
        createFile(path.join(currentPath, name));
    } else {
        createDir(path.join(currentPath, name));
    }
}

console.log("files created successfully...");

File: touch.py
import sys
import os

def createFile(path):
    if not os.path.exists(path):
        with open(path, 'w') as f:
            f.write('')

def createDir(path):
    if not os.path.exists(path):
        os.makedirs(path)

def main():
    args = sys.argv[1:]
    currentPath = os.getcwd()

    for arg in args:
        if arg == "..":
            currentPath = os.path.dirname(currentPath)
        elif '.' in arg:
            createFile(os.path.join(currentPath, arg))
        else:
            createDir(os.path.join(currentPath, arg))

    print("Files and directories created successfully.")

if __name__ == "__main__":
    main()